src/impl/epoll/external_commit_helper.hpp

```diff
@@ -53,22 +57,38 @@ class ExternalCommitHelper {
         FdHolder(FdHolder const&) = delete;
     };
 
-    void listen();
+    class DaemonThread {
```

Comment:
This could be declared in the .cpp file since nothing in the header needs more than a forward declaration.